tests/unit/test_utils.py

```diff
@@ -0,0 +1,312 @@
+        assert (
+            os.path.join(self.SECOND_TEST_DIRECTORY, self.FILE_IN_SECOND_TEST_DIRECTORY)
+            not in zip_file.namelist()
+        )
+
+    def test_recursive_zip_packages_with_env_variable(
+        self,
+        temp_test_directory: str,
+        file_in_a_subdir: str,
+        other_directory: str,
+        file_in_other_directory: str,
+        include_paths_env_variable,
+    ):
+        zip_file_path = os.path.join(temp_test_directory, "packed.zip")
+
+        utils.recursive_zip_packages_dir(temp_test_directory, zip_file_path)
+        zip_file = ZipFile(zip_file_path)
+
+        assert os.path.isfile(zip_file_path)
+        assert (os.path.join(self.SUBDIR, self.FILE_IN_A_SUBDIR)) in zip_file.namelist()
+        assert os.path.join(self.FILE_IN_SECOND_TEST_DIRECTORY) in zip_file.namelist()
+
+    def test_standard_zip_dir(self, temp_test_directory: str, file_in_a_subdir: str):
+        zip_file_path = os.path.join(temp_test_directory, "packed.zip")
+        utils.standard_zip_dir(zip_file_path)
+        zip_file = ZipFile(zip_file_path)
+
+        assert os.path.isfile(zip_file_path)
+        assert (
+            os.path.join("subdir", os.path.basename(file_in_a_subdir))
+            not in zip_file.namelist()
+        )
+
+    # Setup functions
+    # These functions are used to set up files and directories used in tests
+    # and delete them, after the tests are performed
+
+    @pytest.fixture
+    def temp_test_directory(self) -> Generator:
+        print(os.getcwd())
+        path = os.path.join(os.getcwd(), self.TEMP_TEST_DIRECTORY).lower()
+        os.mkdir(path)
+        yield path
+        rmtree(path)  # We delete whole directory in teardown -
+        # so, no need to delete any of the files separately
+
+    @pytest.fixture
+    def temp_directory_for_app_zip(self, temp_test_directory: str) -> Generator:
+        path = os.path.join(temp_test_directory, self.TEMP_DIR_FOR_APP_ZIP)
+        os.mkdir(path)
+        yield path
+
+    @pytest.fixture
+    def correct_app_zip(self, temp_test_directory: str) -> Generator:
+        path = os.path.join(temp_test_directory, self.APP_ZIP)
+        dummy_file = open(path, "w")
+        dummy_file.close()
+        yield path
+
+    @pytest.fixture
+    def correct_requirements_txt(self, temp_test_directory: str) -> Generator:
+        path = os.path.join(temp_test_directory, self.REQUIREMENTS_TXT)
+        with open(path, "w") as req_file:
+            for req in requirements:
+                req_file.writelines(req + "\n")
+        yield path
+
+    @pytest.fixture
+    def streamlit_requirements_txt(self, temp_test_directory: str) -> Generator:
+        path = os.path.join(temp_test_directory, self.REQUIREMENTS_SNOWFLAKE)
+        with open(path, "w") as dummy_file:
+            for req in requirements:
+                dummy_file.writelines(req + "\n")
+        yield path
```

Comment:
Can we abstract this part? It seems that it's repeated few times

Reply:
I moved it to separate function